src/exchange/deposit_handler.cairo

```diff
@@ -144,45 +149,108 @@ mod DepositHandler {
         fn create_deposit(
```

Comment:
missing role checks, contract should import IRoleModule and RoleModule and call it's initializer, role_store should be removed from this contract storage.

Reply:
Also missing global reentrancy